substrate/network-libp2p/src/service.rs

```diff
@@ -796,23 +828,23 @@ fn start_kademlia_discovery<T, To, St, C>(shared: Arc<Shared>, transport: T,
 				// that we don't need to run a timer just for flushing.
 				let _ = shared.network_state.flush_caches_to_disk();
 
-				if shared.network_state.should_open_outgoing_connections() {
+				if shared.network_state.should_open_outgoing_custom_connections() != 0 {
 					future::Either::A(perform_kademlia_query(shared.clone(),
 						transport.clone(), swarm_controller.clone()))
 				} else {
 					// If we shouldn't open connections (eg. we reached
 					// `min_peers`), pretend we did a lookup but with an empty
 					// result.
 					trace!(target: "sub-libp2p", "Bypassing kademlia discovery");
-					future::Either::B(future::ok(Vec::new()))
+					future::Either::B(future::ok(()))
 				}
 			}
 		})
 		.for_each({
 			let shared = shared.clone();
-			move |results| {
-				process_kad_results(shared.clone(), transport.clone(),
-					swarm_controller.clone(), results, &local_peer_id);
+			move |_| {
+				connect_to_nodes(shared.clone(), transport.clone(),
```

Comment:
style: this line is fine to avoid the join. otherwise it should be one line per arg, but that's overkill.